cot/src/config.rs

```diff
@@ -809,7 +830,104 @@ impl SessionMiddlewareConfigBuilder {
         }
     }
 }
+/// The type of email backend to use.
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum EmailBackendType {
+    /// No email backend.
+    #[default]
+    None,
+    /// SMTP email backend.
+    Smtp,
+}
+/// The configuration for the SMTP backend.
+///
+/// This is used as part of the [`EmailBackendConfig`] enum.
+///
+/// # Examples
+///
+/// ```
+/// use cot::config::EmailBackendConfig;
+///
+/// let config = EmailBackendConfig::builder().build();
+/// ```
+#[derive(Debug, Default, Clone, PartialEq, Eq, Builder, Serialize, Deserialize)]
+#[builder(build_fn(skip, error = std::convert::Infallible))]
+#[serde(default)]
+pub struct EmailBackendConfig {
+    /// The type of email backend to use.
+    /// Defaults to `None`.
+    #[builder(setter(into, strip_option), default)]
+    pub backend_type: EmailBackendType,
+    /// The SMTP server host address.
+    /// Defaults to "localhost".
+    #[builder(setter(into, strip_option), default)]
+    pub smtp_mode: email::SmtpTransportMode,
```

Comment:
Please add tests to ensure all variants of this enum can be constructed from TOML.